app/middlewares/token/checkToken.js

```diff
@@ -0,0 +1,16 @@
+import jwt from 'jsonwebtoken';
+import { db_config } from '../../config/db_config';
+
+export const checkToken = (req, res, next) => {
+  if (req.path === '/login') return next();
+
+  const token = req.headers['x-access-token'];
+
+  if (!token) return res.status(401).send({ message: 'No token provided' });
+
+  return jwt.verify(token, db_config.jwt_secret, err => {
+    if (err) return res.status(403).send({ message: 'Failed to authenticate token' });
+
+    return next();

```

Comment:
Do you have to return next() here (and also res above in this code)?

Reply:
Looks like for `next()` we don't need to use `return`, but for `res` i think we should keep it, otherwise `next` will be called even when there is an `error`. As i understand, in this case we don't want it to happen.